pennylane/devices/default_qubit.py

```diff
@@ -547,6 +549,29 @@ def _apply_phase(self, state, axes, parameters, inverse=False):
         phase = self._conj(parameters) if inverse else parameters
         return self._stack([state[sl_0], self._const_mul(phase, state[sl_1])], axis=axes[0])
 
+    def _apply_projector(self, state: TensorLike, operation: Operation):
+        """Applies a projector onto the state along the axis specified by the operation
+        wires.
+
+        Args:
+            state (array[complex]): input state
+            operation (~.Projector): operation to apply onto the state
+
+        Returns:
+            array[complex]: output state
+        """
+        axes = self.wires.indices(operation.wires)
+        projector = operation.data[0]
+
+        ndim = self._ndim(state)
+        sl = _get_slice(int(not projector[0]), axes[0], ndim)
+
+        state[sl] = state[sl] * 0.0
+        if np.allclose(state, 0.0):
+            return np.NaN
+
+        return state / np.linalg.norm(state)
```

Comment:
Do we need to use `qml.math` for differentiability?

Reply:
`DefaultQubit` uses numpy for all operations. The logic for ML interfaces is in the passthrough devices.